### ISSUE TITLE
UnboundLocalError: local variable 'entity_ids' referenced before assignment

### ISSUE THREAD

**Issue by ahaverty** [basnijholt/adaptive-lighting#322]
#### Version information:
v1.0.17

#### Description:
I've just updated to v1.0.17 (PR #321 )
I was having issues reported in #283 and #75 due to my use of 'areas'.
That original issue is no longer being logged, but this new issue with an unitialised entity_ids variable is appearing.
I can see there's a [new IF ](https://github.com/basnijholt/adaptive-lighting/pull/321/files#diff-2ae6ce86e459ad0e8703be113d8bd00c7fb232c57bc6133c2f3e5a680fd008b2R1281)(`if ATTR_ENTITY_ID in service_data:`), that I'm guessing isn't being entered, and therefore means that `entity_ids` is uninitialised in the later IF.

```
Traceback (most recent call last):
  File "/config/custom_components/adaptive_lighting/switch.py", line 1295, in turn_on_off_event_listener
    if not any(eid in self.lights for eid in entity_ids):
UnboundLocalError: local variable 'entity_ids' referenced before assignment
```

**Comment by basnijholt** [comment]
Thanks for reporting!

Do you have any idea how this is triggered? I can push some updates with more dense logging later, such that we can identify the issue.

**Comment by basnijholt** [comment]
I added a fix in https://github.com/basnijholt/adaptive-lighting/pull/324.

It would be really great to see what is printed in your logs now.

I released 1.0.18.

If you see `No entity_ids or area_ids found in service_data` in your logs, please report! 😄

**Comment by ahaverty** [comment]
@basnijholt excellent, tested v1.0.20 there, and don't seem to have any adaptive lighting logs anymore, thanks so much 🙌
